eng/verify-version.ps1

```diff
@@ -0,0 +1,62 @@
+[CmdletBinding()]
+Param(
+    [Parameter(Mandatory)]
+    [string] $VSTestExe
+)
+
+$Script:RootDir = (Get-Item (Split-Path $MyInvocation.MyCommand.Path)).Parent.FullName
+
+function Extract-VSTestVersion {
+    param (
+        [string]$VSTestExe
+    )
+
+    $outputFile = Join-Path $RootDir "output.txt"
+
+    Invoke-Expression $VSTestExe > $outputFile 2>&1
+    $firstLine = Get-Content -Path $outputFile -TotalCount 1
+    Remove-Item $outputFile
+
+    $version = ""
+    $matchFound = $firstLine -match "(?<=Version\s+).*"
+    if ($matchFound) {
+        $version = $matches[0]
+    }
+
+    return $version
+}
+
+# vstest.console.exe version info.
+$Script:VSTestVersion = Extract-VSTestVersion -VSTestExe $VSTestExe
+$Script:VersionIsPreview = $VSTestVersion -like "*preview*"
+$Script:VersionIsRelease = $VSTestVersion -like "*release*"
+$Script:VersionIsRTM = $VSTestVersion -match "^\d+.\d+.\d+( \(x\d+\))?$"
```

Comment:
Escape your dots.

Reply:
I am also not sure what this regex is trying to match in the sub-match part.

Reply:
Done, thanks !

Reply:
Regarding the sub-match, the RTM `vstest.console.exe` version is something like `17.9.0 (x64)`. The second part matches ` (x64)`. Not relevant to the version part but I wasn't sure how to ignore this and not ignore `-preview-YYYYMMDD-BB` or `-release-YYYYMMDD-BB` in case they were present, since that would mean the RTM check should've failed.

Reply:
You would match until a word boundary (or space)
https://regex101.com/r/95iOmQ/1